src/storage/src/render/multi_worker_persist_sink.rs

```diff
@@ -0,0 +1,1123 @@
+    batch_descriptions: &Stream<G, (Antichain<Timestamp>, Antichain<Timestamp>)>,
+    desired_stream: &Stream<G, (Result<Row, DataflowError>, Timestamp, Diff)>,
+    persist_clients: Arc<PersistClientCache>,
+    storage_state: &mut StorageState,
+) -> (Stream<G, HollowBatchAndMetadata>, Rc<dyn Any>)
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let worker_index = scope.index();
+
+    let persist_location = target.persist_location.clone();
+    let shard_id = target.data_shard;
+    let target_relation_desc = target.relation_desc.clone();
+
+    let source_statistics = storage_state
+        .source_statistics
+        .get(&collection_id)
+        .expect("statistics initialized")
+        .clone();
+
+    let mut write_op =
+        AsyncOperatorBuilder::new(format!("{} write_batches", operator_name), scope.clone());
+
+    let (mut output, output_stream) = write_op.new_output();
+
+    let mut descriptions_input = write_op.new_input(&batch_descriptions.broadcast(), Pipeline);
+    let mut desired_input = write_op.new_input(desired_stream, Pipeline);
+
+    // This operator accepts the current and desired update streams for a `persist` shard.
+    // It attempts to write out updates, starting from the current's upper frontier, that
+    // will cause the changes of desired to be committed to persist, _but only those also past the
+    // upper_.
+
+    let shutdown_button = write_op.build(move |_capabilities| async move {
+        // In-progress batches of data, keyed by timestamp.
+        let mut stashed_batches = BTreeMap::new();
+
+        // Contains descriptions of batches for which we know that we can
+        // write data. We got these from the "centralized" operator that
+        // determines batch descriptions for all writers.
+        //
+        // `Antichain` does not implement `Ord`, so we cannot use a `BTreeMap`. We need to search
+        // through the map, so we cannot use the `mz_ore` wrapper either.
+        #[allow(clippy::disallowed_types)]
+        let mut in_flight_batches = std::collections::HashMap::<
+            (Antichain<Timestamp>, Antichain<Timestamp>),
+            Capability<Timestamp>,
+        >::new();
+
+        // TODO(aljoscha): We need to figure out what to do with error results from these calls.
+        let persist_client = persist_clients
+            .open(persist_location)
+            .await
+            .expect("could not open persist client");
+
+        let mut write = persist_client
+            .open_writer::<SourceData, (), Timestamp, Diff>(
+                shard_id,
+                &format!("compute::persist_sink::write_batches {}", collection_id),
+                Arc::new(target_relation_desc),
+                Arc::new(UnitSchema),
+            )
+            .await
+            .expect("could not open persist shard");
+
+        // The current input frontiers.
+        let mut batch_descriptions_frontier = Antichain::from_elem(TimelyTimestamp::minimum());
```

Comment:
In the rest of the codebase whenever `timely::progress::Timestamp` and `mz_repr::Timestamp` coexist we use `Timestamp` for the timely trait and `mz_repr::Timestamp` for the concrete mz timestamp. Can we follow this pattern here too?